vowpalwabbit/parse_args.cc

```diff
@@ -1197,17 +1197,23 @@ void load_input_model(vw& all, io_buf& io_temp)
 
 VW::LEARNER::base_learner* setup_base(options_i& options, vw& all)
 {
-  auto setup_func = all.reduction_stack.top();
+  reduction_setup_fn setup_func = std::get<1>(all.reduction_stack.top());
+  std::string setup_func_name = std::get<0>(all.reduction_stack.top());
   all.reduction_stack.pop();
-  auto base = std::get<1>(setup_func)(options, all);
+
+  // 'hacky' way of keeping track of the option group created by the setup_func about to be created
+  options.tint(setup_func_name);
+  auto base = setup_func(options, all);
+  // reset back to general
+  options.tint("general");
```

Comment:
nit: maybe `options.un_tint` could set `"general"` in `options_boost_po` so that we encapsulate the `general` string in one place

Reply:
yes, I'm not that happy with this tint solution but haven't figured out anything better. untint or reset_tint could also work though

Reply:
hm would something similar to the scope_guards logic work? create a tint object which resets to "general" on destruction? (might be an overkill)

Reply:
that feels like the using keyword on c#

Reply:
semi-done